java/arcs/core/data/proto/ParticleSpecProtoDecoder.kt

```diff
@@ -35,7 +36,7 @@ fun HandleConnectionSpecProto.decode() = HandleConnectionSpec(
     name = name,
     direction = direction.decode(),
     type = type.decode(),
-    expression = expression.ifEmpty { null }
+    expression = if (expression.isEmpty()) { null } else { PaxelParser.parse(expression) }
```

Comment:
I really wish there was a String.ifNotEmpty { ... }. I was trying to think of a clever way to write this but its just as verbose, e.g.

```
expression.ifEmpty { null }?.takeIf { PaxelParser.parse(expression) } 
```

but I don't think it's better.